pkg/controller/cluster/main-controller.go

```diff
@@ -830,6 +828,25 @@ func (c *Controller) syncHandler(key string) error {
 		}
 	}
 
+	// Handle the ClusterIP Service for Tenant Console
+	_, err = c.serviceLister.Services(tenant.Namespace).Get(tenant.ConsoleCIServiceName())
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			if tenant, err = c.updateTenantStatus(ctx, tenant, StatusProvisioningConsoleService, 0); err != nil {
+				return err
+			}
+			klog.V(2).Infof("Creating a new ClusterIP Service for Console %q", nsName)
+			// Create the headless service for the tenant
+			consoleSvc := services.NewClusterIPForConsole(tenant)
+			_, err = c.kubeClientSet.CoreV1().Services(tenant.Namespace).Create(ctx, consoleSvc, cOpts)
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
```

Comment:
please add logic to compare the current state of the service to the spec, in case they are passing new labels/annotations to the service